src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -122,47 +134,196 @@ class InlineTrans(Transformation):
         Some of these restrictions will be lifted by #924.
 
     '''
-    def apply(self, node, options=None):
-        '''
+
+    def __init__(self):
+        # List of call-to-subroutine argument indices
+        self._ret_arg_match_list: List[int] = None
```

Comment:
Why is this Transformation class now holding state? I wouldn't expect this to be necessary...